signer/tests/integration/transaction_signer.rs

```diff
@@ -161,6 +203,151 @@ async fn should_be_able_to_participate_in_signing_round() {
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[test(tokio::test)]
+async fn should_store_sweep_transaction_info_from_other_signers() {
+    let num_signers = 3;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(46);
+    let network = InMemoryNetwork::new();
+    let signer_info = testing::wsts::generate_signer_info(&mut rng, num_signers);
+    let signer_set_pubkeys = &signer_info.first().unwrap().signer_public_keys;
+    let mut coord_network = network.connect();
+
+    // Instantiate a new database for each signer. This must be done sequentially,
+    // it fails if done concurrently.
+    let mut signer_dbs: Vec<PgStore> = vec![];
+    for _ in signer_info.iter() {
+        signer_dbs.push(create_signer_database().await);
+    }
+
+    // A closure to build a new context for each signer which will use one of the
+    // databases pre-created above.
+    let build_context = |index: usize, private_key| {
+        TestContext::builder()
+            .with_storage(signer_dbs[index].clone())
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = private_key;
+                settings.signer.bootstrap_signing_set =
+                    signer_set_pubkeys.iter().cloned().collect();
+            })
+            .build()
+    };
+
+    // Create a new event-loop for each signer, based on the number of signers
+    // defined in `self.num_signers`. Note that it is important that each
+    // signer has its own context (and thus storage and signalling channel).
+    //
+    // Each signer also gets its own `MpscBroadcaster` instance, which is
+    // backed by the `network` instance, simulating a network connection.
+    let signers: Vec<_> = signer_info
+        .iter()
+        .enumerate()
+        .map(|(index, signer_info)| {
+            let context = build_context(index, signer_info.signer_private_key);
+            TxSignerEventLoop {
+                network: network.connect(),
+                context: context.clone(),
+                context_window: 10000,
+                blocklist_checker: Some(()),
+                wsts_state_machines: HashMap::new(),
+                signer_private_key: context.config().signer.private_key,
+                threshold: 2,
+                rng: rand::rngs::StdRng::seed_from_u64(51),
+            }
+        })
+        .collect();
+
+    // Generate test data. We'll generate two blocks and include all outstanding
+    // deposit and withdraw requests in the sweep transaction info we broadcast.
+    let test_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 2,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 2,
+        num_withdraw_requests_per_block: 2,
+        num_signers_per_request: 0,
+    };
+    let test_data = TestData::generate(&mut rng, &[], &test_params);
+    let bitcoin_chain_tip = test_data.bitcoin_blocks.last().unwrap().block_hash;
```

Comment:
nit: this is (always) the correct chain tip only for `num_bitcoin_blocks` <= 2, to avoid surprises should someone change that number what about using one of the storages to get the chain tip (after we populate them)?

Reply:
yeah that's a good point, updated in [f66e1fa](https://github.com/stacks-network/sbtc/pull/732/commits/f66e1fa097f684580fc0c3cf42bb56cf5e806bb8)